### DISCUSSION TITLE
Branch permissions for outside collaborator

### DISCUSSION THREAD

**Discussion by ro0NL** [community/community#64234] [Repositories]
### Select Topic Area

Question

### Body

Hi,

I've added an outside collaborator to a specific (private) repository, with READ permissions.

I want this user to be able to ONLY push in a designated branch we've created.

To me this sounds like a common usecase, but it isnt obvious to me how it should be configured in github.

My idea was to enable a branch protection rule for this designated branch, and then select the outside collaborator user under "Restrict who can push to matching branches", given:

> Specify people, teams, or apps allowed to push to matching branches.

However the select menu only shows team MEMBERS, not any "people"/outside collaborators.

Am i misunderstanding it? Can the usecase be achieved otherwise?

**Comment by Jaber-Saed** [comment]
Hello,

Your approach is correct, but you're facing a limitation in the GitHub interface when trying to add outside collaborators to branch protection rules. Currently, GitHub's branch protection rules interface only allows you to select team members and not individual outside collaborators when configuring who can push to matching branches.

However, there's a workaround you can use to achieve the desired outcome:

1. **Create a Team**: You can create a team specifically for this outside collaborator, even if they are the only member of the team.

2. **Add the Outside Collaborator to the Team**: Add the outside collaborator to the team you created. Even though they are an individual outside collaborator, placing them in a team allows you to work around the limitation of GitHub's interface.

3. **Branch Protection Rule**: Now, when you set up the branch protection rule, you can select the team you created to be allowed to push to the designated branch.

This way, you're using the team as an intermediary to assign permissions to the individual outside collaborator for that specific branch. It might seem a bit convoluted, but it's a common workaround used to handle situations like yours where you want to provide specific permissions to an outside collaborator for a specific branch.

Remember to make sure the rest of the team settings, like permissions and access, are configured according to your requirements to maintain proper security and access controls.

Keep in mind that GitHub's interface might change over time, so it's always a good idea to check their official documentation or support resources for the most up-to-date information on managing permissions and branch protections.

I hope this helps! If you have any more questions or need further assistance, feel free to ask.
